include/diaspora.php

```diff
@@ -828,17 +828,17 @@ private static function post_allow($importer, $contact, $is_comment = false) {
 			logger("defining user ".$contact["nick"]." as friend");
 		}
 
-		if (($contact["blocked"]) || ($contact["readonly"]) || ($contact["archive"]))
+		if ($contact["blocked"] || $contact["readonly"] || $contact["archive"]) {
 			return false;
-		if ($contact["rel"] == CONTACT_IS_SHARING || $contact["rel"] == CONTACT_IS_FRIEND)
+		} elseif (($contact["rel"] == CONTACT_IS_SHARING) || ($contact["rel"] == CONTACT_IS_FRIEND)) {
 			return true;
-		if ($contact["rel"] == CONTACT_IS_FOLLOWER)
-			if (($importer["page-flags"] == PAGE_COMMUNITY) OR $is_comment)
-				return true;
-
-		// Messages for the global users are always accepted
-		if ($importer["uid"] == 0)
+		} elseif (($contact["rel"] == CONTACT_IS_FOLLOWER) && ($importer["page-flags"] == PAGE_COMMUNITY)) {
```

Comment:
Can't you combine this condition with the previous `elseif` that also `return true;`?